docs/data/tree-view/rich-tree-view/editing/Validation.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Tooltip from '@mui/material/Tooltip';
+import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { useTreeItem2Utils } from '@mui/x-tree-view/hooks';
+import { TreeItem2, TreeItem2Props } from '@mui/x-tree-view/TreeItem2';
+import { UseTreeItem2LabelInputSlotProps } from '@mui/x-tree-view/useTreeItem2/useTreeItem2.types';
+import { TreeItem2LabelInput } from '@mui/x-tree-view/TreeItem2LabelInput';
+import { MUI_X_PRODUCTS } from './products';
+
+const ERRORS = {
+  REQUIRED: 'The label cannot be empty',
+  INVALID: 'The label cannot contain digits',
+};
+
+interface CustomLabelInputProps extends UseTreeItem2LabelInputSlotProps {
+  error: null | keyof typeof ERRORS;
```

Comment:
Did you check if TS was able to correctly get the keys.

I would expect you to need to use `as const`

```tsx
const ERRORS = {
  REQUIRED: 'The label cannot be empty',
  INVALID: 'The label cannot contain digits',
} as const;
```

Reply:
![image](https://github.com/user-attachments/assets/feeba921-9c1b-42e3-b883-16c8578bcd78)
I think it gets them correctly? 🤔

Reply:
Looks like it
Maybe TS got smarter recently